toolchain/check/type_completion.cpp

```diff
@@ -544,46 +551,68 @@ auto RequireConcreteType(Context& context, SemIR::TypeId type_id,
   return true;
 }
 
-auto RequireDefinedType(Context& context, SemIR::TypeId type_id,
-                        SemIR::LocId loc_id,
-                        Context::BuildDiagnosticFn diagnoser) -> bool {
-  if (!RequireCompleteType(context, type_id, loc_id, diagnoser)) {
-    return false;
-  }
-
-  if (auto facet_type = context.types().TryGetAs<SemIR::FacetType>(type_id)) {
-    const auto& facet_type_info =
-        context.facet_types().Get(facet_type->facet_type_id);
-    for (auto interface : facet_type_info.impls_constraints) {
-      auto interface_id = interface.interface_id;
-      if (!context.interfaces().Get(interface_id).is_defined()) {
-        auto builder = diagnoser();
-        NoteUndefinedInterface(context, interface_id, builder);
-        builder.Emit();
-        return false;
-      }
+static auto RequireCompleteFacetTypeImpl(
```

Comment:
```suggestion
static auto AddCompleteFacetType(
```
Since "impl" means stuff in the language, I have been trying to avoid using it as a suffix in this way in the toolchain, so suggesting an alternative phrasing if you like it.